document/js-api/index.bs

```diff
@@ -0,0 +1,840 @@
+    1. Let |values| be a list whose length is [=size_table=](|store|, |tableaddr|) where each element is null.
+    1. Return a new {{Table}} instance with \[[Table]] set to |tableaddr| and \[[Values]] set to |values|.
+</div>
+
+<div algorithm>
+    The <dfn constructor for="Table">Table(descriptor)</dfn> constructor, when invoked, performs the following steps:
+    1. Let |n| be |descriptor|.initial.
+    1. If |descriptor|.maximum is present, let |m| be |descriptor|.maximum; otherwise, let |m| be empty.
+    1. If |m| is not empty and |m| &lt; |n|, throw a {{RangeError}} exception.
+    1. Let |type| be the [=table type=] {𝗆𝗂𝗇 n, 𝗆𝖺𝗑 |m|} 𝖺𝗇𝗒𝖿𝗎𝗇𝖼.
+    1. Let |store| be the current agent cluster's [=associated store=].
+    1. Let (|store|, |tableaddr|) be [=alloc_table=](|store|, |type|). <!-- TODO(littledan): Report allocation failure https://github.com/WebAssembly/spec/issues/584 -->
+    1. Set the current agent cluster's [=associated store=] to |store|.
+    1. [=Create a table object=] from the table address |tableaddr| and return the result.
+</div>
+
+<div algorithm>
+    The <dfn method for="Table">grow(d)</dfn> method, when invoked, performs the following steps:
+    1. Let |tableaddr| be the Table instance's \[[Table]] internal slot.
+    1. Let |initialSize| be the length of the Table instance's \[[Values]] internal slot.
+    1. Let |store| be the current agent cluster's [=associated store=].
+    1. Let |result| be [=grow_table=](|store|, |tableaddr|, |d|).
+    1. If |result| is [=error=], throw a {{RangeError}} exception.
+
+        Note: The above exception may happen due to either insufficient memory or an invalid size parameter.
+
+    6. Set the current agent cluster's [=associated store=] to |store|.
+    1. [=Append=] null to the Table instance's \[[Values]] internal slot |d| &minus; |initialSize| times.
+    1. Return |initialSize|.
+</div>
+
+<div algorithm>
+  The getter of the <dfn attribute for="Table">length</dfn> attribute of {{Table}} returns the length of the table's \[[Values]] internal slot.
+</div>
+
+<div algorithm>
+    The <dfn method for="Table">get(index)</dfn> method, when invoked, performs the following steps:
+    1. Let |values| be the Table instance's \[[Values]] internal slot.
+    1. Let |size| be the length of |values|.
+    1. If |index| &ge; |size|, throw a {{RangeError}} exception.
+    1. Return |values|[|index|].
+</div>
+
+<div algorithm>
+    The <dfn method for="Table">set(index, value)</dfn> method, when invoked, performs the following steps:
+    1. Let |tableaddr| be the Table instance's \[[Table]] internal slot.
+    1. Let |values| be the Table instance's \[[Values]] internal slot.
+    1. If |value| is null, let |elem| be an empty [=function element=].
+    1. Otherwise,
+        1. If |value| does not have a \[[FunctionAddress]] internal slot, throw a {{TypeError}} exception.
+        1. Let |funcaddr| be |value|.\[[FunctionAddress]].
```

Comment:
I think it should say "Let `elem` be `value`.[[FunctionAddress]].

Reply:
Fixed (using funcaddr rather than elem as the name)